core/src/main/scala/org/http4s/util/Logging.scala

```diff
@@ -0,0 +1,118 @@
+package org.http4s.util
+
+import cats._
+import cats.implicits._
+import cats.data.Kleisli
+import cats.effect.{Effect, Sync}
+import org.http4s.Method.NoBody
+import org.http4s._
+//import org.http4s.Method._
+import org.log4s.Logger
+import org.log4s.LogLevel
+import fs2._
+
+import scala.concurrent.ExecutionContext
+
+object Logging {
+
+
+  def logMessage[F[_], A <: Message[F]](
+                                        f: Kleisli[F, A, String],
+                                        cacheBody: A => Boolean = {_: A => true},
+                                        forceBody: A => Boolean = {_: A => false},
+                                        logLevel: A => LogLevel = {_: A => org.log4s.Trace},
+                                        a: A,
```

Comment:
I'd put things without defaults ahead of things with defaults.

Reply:
This was originally curried and I put it back in at some point. I think in general it should probably be curried. As that leaves it as a function that works as a logger taking the thing that does the logger action and the thing to log.